### DISCUSSION TITLE
Neper Multi-level Boundary Model Construction

### DISCUSSION THREAD

**Discussion by fine-hhh** [neperfepx/neper#935] [Q&A]
I want to generate the following microstructure using Neper, with the grain boundaries having thickness. However, when I use the following command, I get an error saying that the mesh is discontinuous. What should I do?
```
neper -T -n "10::file(refined.txt)" -id 1::1 -dim 2 -morpho "centroidal" -transform "scale(0.07,0.07)" -domain "square(80,80)" -format tess,ori -o n1
neper -M n1.tess -interface cohesive -format inp -o n1
neper -M n1.tess -interface cohesive -format inp -o n1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.10.1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 2 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2024, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/hhh/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n1.tess -interface cohesive -format inp -o n1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n1.tess'...
Info   :     [i] Parsed file `n1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.2214) 
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing... 100% (0.79|0.89/80%|10%|10%) 
Info   :   - Processing mesh at interfaces...
Info   :     > Making mesh discontinuous...
已中止 (核心已转储)
```

![65588e34f4f2745d13fe554402bc01f](https://github.com/user-attachments/assets/138d44fe-edb1-4ba8-9cdc-20de0cb5de48)

![n1_image](https://github.com/user-attachments/assets/308f8a4e-15a6-4850-afec-e4969a3212f6)

**Comment by rquey** [comment]
With version 4.10.2-2, this should work:
```
neper -T -n "10::file(refined.txt)" -id 1::1 -dim 2 -morpho "centroidal" -transform "scale(0.07,0.07)" -domain "square(80,80)" -format tess,ori -o n1
neper -M n1.tess -interface cohesive -format inp -o n1 -order 1
```

**Reply by fine-hhh** [reply]
Thank you for your reply. I tried again using your instructions, but the mesh is still discontinuous.
 neper -M n1.tess -interface cohesive -format inp -o n1 -order 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.10.1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 2 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2024, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/hhh/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n1.tess -interface cohesive -format inp -o n1 -order
         1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n1.tess'...
Info   :     [i] Parsed file `n1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.2214) 
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing... 100% (0.79|0.89/80%|10%|10%) 
Info   :   - Processing mesh at interfaces...
Info   :     > Making mesh discontinuous...
已中止 (核心已转储)

**Reply by rquey** [reply]
Update your Neper.
